core/src/main/java/net/opentsdb/query/execution/serdes/JsonV3QuerySerdes.java

```diff
@@ -511,17 +525,19 @@ private void writeNumericSummary(
 
     if (result.timeSpecification() != null) {
 
-      final List<Integer> summaries = Lists.newArrayList(
-          ((TimeSeriesValue<NumericSummaryType>) value).value().summariesAvailable());
-
-      if (summaries != null && summaries.size() == 1) {
-        // If the number of summaries are 1, going to force a NumericType on the data
-        // TODO: find a clean way of doing this
+      if (!(result.source() instanceof Summarizer)) {
         writeRollupNumeric((TimeSeriesValue<NumericSummaryType>) value, options, iterator, json,
             result);
         return;
       }
 
+      Collection<Integer> integers =
+          ((TimeSeriesValue<NumericSummaryType>) value)
+              .value()
+              .summariesAvailable();
+
+      final List<Integer> summaries = Lists.newArrayList(integers);
```

Comment:
Why the dupe here? This may be super expensive as it would create a new list for every data point